pkg/deployments/scaler.go

```diff
@@ -14,13 +14,37 @@ type scaler struct {
 	minScale     int32
 	maxScale     int32
 
+	lastScaleDown time.Time
+
 	// scaleFuncMtx ensures the scale function is not run concurrently.
 	scaleFuncMtx sync.Mutex
 	scaleFunc    func(n int32, atLeastOne bool) error
 
-	scaleDownDelay   time.Duration
-	scaleDownStarted bool
-	scaleDownTimer   *time.Timer
+	scaleDownDelay time.Duration
+}
+
+func newScaler(scaleDownDelay time.Duration, scaleFunc func(int32, bool) error) *scaler {
```

Comment:
In future, it would be nice to keep the functions in place to avoid the noisy diff (its hard to know exactly what was changed here)... We can do re-ordering in another PR that does not change logic.

Reply:
ok will do. I will do 2 PRs and then base 2nd PR on the 1st refactor PR that doesn't change logic